RoslynPluginGenerator/Program.cs

```diff
@@ -14,22 +15,24 @@ namespace SonarQube.Plugins.Roslyn
     /// </summary>
     public static class Program
     {
-        public const string NuGetPackageSource = "https://www.nuget.org/api/v2/";
-
         private const int ERROR_CODE = 1;
         private const int SUCCESS_CODE = 0;
 
         static int Main(string[] args)
         {
             ConsoleLogger logger = new ConsoleLogger();
-            Common.Utilities.LogAssemblyVersion(typeof(Program).Assembly, UIResources.AssemblyDescription, logger);
+            Utilities.LogAssemblyVersion(typeof(Program).Assembly, UIResources.AssemblyDescription, logger);
             
             ProcessedArgs processedArgs = ArgumentProcessor.TryProcessArguments(args, logger);
 
             bool success = false;
             if (processedArgs != null)
             {
-                NuGetPackageHandler packageHandler = new NuGetPackageHandler(NuGetPackageSource, logger);
+                ISettings nuGetSettings = NuGetRepositoryFactory.GetSettingsFromConfigFiles();
+                IPackageRepository repo = NuGetRepositoryFactory.CreateRepository(nuGetSettings, logger);
+                string localNuGetCache = Utilities.CreateTempDirectory(".nuget");
+                NuGetPackageHandler packageHandler = new NuGetPackageHandler(repo, localNuGetCache, logger);
```

Comment:
The Program is now responsible for creating the appropriate repository and deciding where the local cache should be. Previously this was hidden in the NuGetPackageHandler constructors. However, the constructors were getting a bit complicated.